File: data/src/main/java/com/marvelcomics/brito/data/datasource/remote/response/model/RemoteMarvelContainer.java
package com.marvelcomics.brito.data.datasource.remote.response.model;

import com.google.gson.annotations.SerializedName;

public class RemoteMarvelContainer<T> {

    @SerializedName("code")
    private int code;
    @SerializedName("status")
    private String status;
    @SerializedName("copyright")
    private String copyright;
    @SerializedName("attributionText")
    private String attributionText;
    @SerializedName("attributionHTML")
    private String attributionHTML;
    @SerializedName("etag")
    private String etag;
    @SerializedName("data")
    private RemoteMarvelData<T> remoteMarvelData;

    public RemoteMarvelData<T> getRemoteMarvelData() {
        return remoteMarvelData;
    }
}

File: data/src/main/java/com/marvelcomics/brito/data/datasource/remote/response/model/RemoteMarvelContainer.kt
package com.marvelcomics.brito.data.datasource.remote.response.model

import com.google.gson.annotations.SerializedName

class RemoteMarvelContainer<Type> {
    @SerializedName("code")
    private val code = 0
    @SerializedName("status")
    private val status: String? = null
    @SerializedName("copyright")
    private val copyright: String? = null
    @SerializedName("attributionText")
    private val attributionText: String? = null
    @SerializedName("attributionHTML")
    private val attributionHTML: String? = null
    @SerializedName("etag")
    private val etag: String? = null
    @SerializedName("data")
    val remoteMarvelData: RemoteMarvelData<Type>? = null
}